store/sqlstore/post_store.go

```diff
@@ -334,7 +334,7 @@ func (s *SqlPostStore) InvalidateLastPostTimeCache(channelId string) {
 
 func (s *SqlPostStore) GetEtag(channelId string, allowFromCache bool) string {
 	if allowFromCache {
-		if cacheItem, ok := s.lastPostTimeCache.Get(channelId); ok && cacheItem.(int64) > 0 {
+		if cacheItem, ok := s.lastPostTimeCache.Get(channelId); ok {
```

Comment:
[HH comment](https://github.com/mattermost/mattermost-server/pull/10861#discussion_r286046656): What's the reason for this change? I don't understand why that extra check was there in the first place, but I'm not sure why it would we removed

Reply:
I initially got a failing test and I thought it was related to it.  Reverted on last commit - https://github.com/mattermost/mattermost-server/pull/11039/commits/fe6bfe81f280a3d6746e1c99e9a0e7093217eea8.
The reason I added that before was that there's instance the cacheItem is equal to `0` and in turn, it returns outdated posts.  Honestly, I was not able to replicate the original issue since there's no client yet that's using this unread API.  That extra check wouldn't hurt but I'll keep an eye about the original issue.

Reply:
Okay, this is fine then